### PR TITLE
ui/api: Prohibit root from creating subprojects

### LINKED ISSUE #378: ui/api: Prohibit root from adding subprojects
Follow up to #39

### PR DESCRIPTION
**Pull request by miamarina**
# Changes

* Disable create subproject button for root
* Return precondition error in api if root tries
to create a subproject
* Add subproject_create_spec.js and tests in subproject_create.spec.ts

# How to test

Test subproject_create_spec.js and tests in subproject_create.spec.ts have been added

Closes #378

### CODE REVIEW

File: api/src/service/domain/workflow/subproject_create.spec.ts [hunk 1]
```diff
@@ -70,4 +70,24 @@ describe("create subproject & projected budgets", () => {
...
+        { organization: "orga2", currencyCode: "EUR", value: "2" },
+      ],
+    };
+
+    const result = await SubprojectCreate.createSubproject(ctx, rootuser, data, {
+      subprojectExists: async (_projectId, _subprojectId) => false,
+      projectPermissions: async _projectId => ({ "project.createSubproject": [rootuser.id] }),
+    });
+
+    assert.isOk(Result.isErr(result));
```
Comment (mathiashoeld) [inline api/src/service/domain/workflow/subproject_create.spec.ts]: please use 
``` 
assert.isTrue(Result.isErr(result));
```
instead

File: api/src/service/domain/workflow/subproject_create.spec.ts [hunk 2]
```diff
@@ -70,4 +70,24 @@ describe("create subproject & projected budgets", () => {
...
+  });
```
Comment (mathiashoeld) [inline api/src/service/domain/workflow/subproject_create.spec.ts]: add `assert.instanceOf(...)` to check for the correct error (PreconditionError)

File: e2e-test/cypress/integration/subproject_create_spec.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+describe("Create a subproject", function() {
```
Comment (Stezido) [inline e2e-test/cypress/integration/subproject_create_spec.js]: Due to the other describe messages I suggest a similar format
``` suggestion
describe("Subproject creation", function() {
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -17,7 +17,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - If a user changes his/her password, the new password has to follow security guidelines [#370](https://github.com/openkfw/TruBudget/issues/370)
 - Allow user 'root' to change the passwords of all users[#366](https://github.com/openkfw/TruBudget/issues/366)
 - Disallow root to create projects [#375](https://github.com/openkfw/TruBudget/issues/375)
-- Allow user 'root' to change the passwords of all users[#366](https://github.com/openkfw/TruBudget/issues/366)
+<<<<<<< HEAD
+- Disallow root to create subprojects [#378](https://github.com/openkfw/TruBudget/issues/378)
+- Disallow root to create subprojects [#378](https://github.com/openkfw/TruBudget/issues/378)
+=======
+>>>>>>> bad20999... ui/api: Prohibit root from adding workflowitems
+- Disallow root to add workflowitems [#380](https://github.com/openkfw/TruBudget/issues/380)
```
Comment (mathiashoeld) [inline CHANGELOG.md]: merge conflict, please resolve

**Pull request merged**